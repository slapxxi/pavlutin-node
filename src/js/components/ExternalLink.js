import React from 'react';

function ExternalLink({ href, children }) {
  return (
    <a
      href={href}
      rel="noopener noreferrer"
      target="_blank"
    >
      { children }
    </a>
  );
}

export default ExternalLink;
