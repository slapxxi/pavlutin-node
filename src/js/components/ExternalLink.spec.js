import React from 'react';
import render from 'react-test-renderer';
import ExternalLink from './ExternalLink';

it('renders link to an external resource', () => {
  const tree = render.create(
    <ExternalLink href="https://poogle.com">Poogle</ExternalLink>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
