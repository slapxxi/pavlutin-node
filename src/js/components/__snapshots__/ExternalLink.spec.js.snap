// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link to an external resource 1`] = `
<a
  href="https://poogle.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Poogle
</a>
`;
